### ISSUE TITLE
Incorrect constructor result for static inner class

### ISSUE THREAD

**Issue by jwaataja** [t-rasmud/checker-framework#114]
Consider this code,
```java
import org.checkerframework.checker.determinism.qual.*;

public class InnerClass {

    public static void f() {
        @Det C a = new C();
    }

    public static class C {
    }
}
```

This gives the following error
```
InnerClass.java:6: error: [assignment.type.incompatible] incompatible types in assignment.
        @Det C a = new C();
                   ^
  found   : @NonDet C
  required: @Det C
```

If a constructor has no `@PolyDet` arguments, its return type should be `@Det` by default but here we see this returns `@NonDet`. This only seems to occur with inner classes, with the following change,

```java
import org.checkerframework.checker.determinism.qual.*;

public class InnerClass {

    public static void f() {
        @Det InnerClass a = new InnerClass();
    }

    public static class C {
    }
}
```

I get no error.

**Comment by smillst** [comment]
https://github.com/typetools/checker-framework/pull/2504 fixes the problem with static inner classes.     Also, some of the code in the Determinism checker can be cleaned up as it was really a work around the problem fixed by that pull request.  I'm working on that now.

**Comment by t-rasmud** [comment]
Fixed in 43726d3912ec359b8f8b6a89f4bd6163a4b6a05c
